src/video.cpp

```diff
@@ -191,6 +191,9 @@ void CVideo::update_framebuffer()
 	}
 
 	surface fb = SDL_GetWindowSurface(*window);
+    if (fb.get() == NULL) {
+        fprintf(stderr, "SDL_GetWindowSurface failed: %s\n", SDL_GetError());
```

Comment:
Please use Wesnoth's logging streams instead of the `printf` functions. (Also this is indented by spaces. If you removed it later, that doesn't matter, but the main point here still applies to any other logging you feel a need to add.)